### DISCUSSION TITLE
xUnit is using a previous iteration to evaluate a different case when using Theory and InlineData

### DISCUSSION THREAD

**Discussion by EduardoSantos7** [xunit/xunit#2500] [Question] [answered]
I'm trying to test a method and verify whether another method was called or not. This depends on a bool value. If it's true, the other method should be called. Still, the assertion occasionally works because the true scenario might run before the false scenario, and xUnit counts the call that happens there.
```csharp
        [Theory]
        [InlineData(false)]
        [InlineData(true)]
        public async Task Test_Resync(bool shouldUpdate)
        {
            ...

            await Resync().ConfigureAwait(false);

            if (shouldUpdate)
            {
                A.CallTo(() => mock.AnyMethod()).MustHaveHappened();
                return;
            }

            A.CallTo(() => mock.AnyMethod()).MustNotHaveHappened();
        }
```
Although I can use MustHaveHappenedOnceOrLess as a workaround, I think this is not reflecting the behavior expected, so my doubt is: is there a way to have independent cases when using Theory? Would it be better to separate both scenarios?

**Comment by bartelink** [comment, marked as answer]
It all depends on what you're doing; up to you. Note that the Test Class is recreated for every InlineData item so not sure what you mean by it working depending on the order - are you using `static` state somewhere (that's rarely a good idea, if you are)
